packages/portalnetwork/src/networks/history/util.ts

```diff
@@ -294,13 +295,60 @@ export const verifyPreCapellaHeaderProof = (
     'blockHash',
   ])
   const reconstructedBlock = ssz.bellatrix.BeaconBlock.createFromProof({
-    witnesses: proof.beaconBlockHeaderProof,
+    witnesses: proof.beaconBlockProof,
     type: ProofType.single,
     gindex: elBlockHashPath.gindex,
     leaf: elBlockHash,
   })
 
-  if (!equalsBytes(reconstructedBlock.hashTreeRoot(), proof.beaconBlockHeaderRoot)) return false
+  if (!equalsBytes(reconstructedBlock.hashTreeRoot(), proof.beaconBlockRoot)) return false
+  return true
+}
+
+export const verifyPostCapellaHeaderProof = (
+  proof: ValueOfFields<{
+    beaconBlockProof: VectorCompositeType<ByteVectorType>
+    beaconBlockRoot: ByteVectorType
+    historicalSummariesProof: VectorCompositeType<ByteVectorType>
+    slot: UintBigintType
+  }>,
+  elBlockHash: Uint8Array,
+  historicalSummaries: { blockSummaryRoot: Uint8Array; stateSummaryRoot: Uint8Array }[],
+  chainConfig: ForkConfig,
+) => {
+  const eraIndex = slotToHistoricalBatchIndex(proof.slot)
+  const forkName = chainConfig.getForkName(Number(proof.slot))
+  const historicalSummariesPath = ssz[forkName].BeaconState.fields.blockRoots.getPathInfo([
+    Number(eraIndex),
+  ])
+  const reconstructedBatch = ssz[forkName].BeaconState.fields.blockRoots.createFromProof({
+    witnesses: proof.historicalSummariesProof,
+    type: ProofType.single,
+    gindex: historicalSummariesPath.gindex,
+    leaf: proof.beaconBlockRoot, // This should be the leaf value this proof is verifying
+  })
+
+  if (
+    !equalsBytes(
+      reconstructedBatch.hashTreeRoot(),
+      historicalSummaries[Number(slotToHistoricalBatch(proof.slot)) - 758].blockSummaryRoot,
```

Comment:
probably store 758 in some constant for later clarity/recall

Reply:
Good call.  Have done.